### PR TITLE
Using NOHW AES-GCM implementation for EDP environment

### PR DESCRIPTION
**Pull request by jimmyyip-crypto**

### CODE REVIEW

**Comment by trevor-crypto** [comment]
Adding @devashishdxt so we can get this merged, and then hopefully pull from a more recent version of `ring`. We're currently hard-stuck on `0.16.15`.

File: src/aead/aes.rs [hunk 1]
```diff
@@ -152,7 +152,7 @@ impl Key {
                 set_encrypt_key!(GFp_vpaes_set_encrypt_key, bytes, key_bits, &mut key)?
             }
 
-            #[cfg(all(not(target_arch = "aarch64"), not(target_env = "sgx")))]
+            #[cfg(all(not(target_arch = "aarch64")))]
```
Comment (devashishdxt) [inline src/aead/aes.rs:155]:
```suggestion
            #[cfg(not(target_arch = "aarch64"))]
```

You can also remove `all()` from every condition as we're only using one condition now.

**Review by tomtau — COMMENTED** [review]
wouldn't you want to use the HW AES instructions? They should generally be available on all CPUs with SGX?

**Comment by tomtau** [comment]
https://github.com/crypto-com/ring/pull/6#pullrequestreview-547897447

**Pull request closed without merging**